src/java/org/apache/cassandra/cql3/statements/CreateRoleStatement.java

```diff
@@ -52,7 +54,11 @@ public class CreateRoleStatement extends AuthenticationStatement
     public CreateRoleStatement(RoleName name, RoleOptions options, DCPermissions dcPermissions,
                                CIDRPermissions cidrPermissions, boolean ifNotExists)
     {
-        this.role = RoleResource.role(name.getName());
+        if (name == null || name.getName() == null)
```

Comment:
For my education, can either of these two happen? What are the different sources for these two different approaches to a name not being present?

Reply:
Also, what about checking for `options. isGeneratedName()` instead? I think it is a more robust approach?

Reply:
yes that is better